hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListenerTokenTest.java

```diff
@@ -791,6 +795,118 @@ void nftUpdateTreasury() {
+                .setSenderAccountID(protoOldTreasury)
+                .build();
+        var nftTreasuryUpdateTransferList = TokenTransferList.newBuilder()
+                .setToken(protoTokenId)
+                .addNftTransfers(nftTreasuryUpdate)
+                .build();
+        var nftUpdateRecordItem = recordItemBuilder.tokenUpdate()
+                .transactionBody(b -> b.setToken(protoTokenId).setTreasury(protoNewTreasury))
+                .record(r -> r.addTokenTransferLists(nftTreasuryUpdateTransferList))
+                .build();
+
+        parseRecordItemsAndCommit(List.of(nftMintRecordItem, nftTransferRecordItem, nftUpdateRecordItem));
+
+        // then
+        nft1.setAccountId(newTreasury);
+        nft1.setModifiedTimestamp(nftUpdateRecordItem.getConsensusTimestamp());
+        nft2.setAccountId(account);
+        nft2.setModifiedTimestamp(nftTransferRecordItem.getConsensusTimestamp());
+        var nft3 = Nft.builder()
+                .id(new NftId(3L, tokenId))
+                .accountId(newTreasury)
+                .createdTimestamp(nftMintRecordItem.getConsensusTimestamp())
+                .deleted(false)
+                .metadata(DomainUtils.toBytes(nftMintRecordItem.getTransactionBody().getTokenMint().getMetadata(0)))
+                .modifiedTimestamp(nftUpdateRecordItem.getConsensusTimestamp())
+                .build();
+        assertThat(nftRepository.findAll()).containsExactlyInAnyOrder(nft1, nft2, nft3);
+
+        var nftTransfer1 = com.hedera.mirror.common.domain.token.NftTransfer.builder()
+                .id(new NftTransferId(nftUpdateRecordItem.getConsensusTimestamp(), 1L, tokenId))
+                .senderAccountId(oldTreasury)
+                .receiverAccountId(newTreasury)
+                .build();
+        var nftTransfer2 = com.hedera.mirror.common.domain.token.NftTransfer.builder()
+                .id(new NftTransferId(nftTransferRecordItem.getConsensusTimestamp(), 2L, tokenId))
+                .senderAccountId(oldTreasury)
+                .receiverAccountId(account)
+                .build();
+        var nftTransfer3 = com.hedera.mirror.common.domain.token.NftTransfer.builder()
+                .id(new NftTransferId(nftMintRecordItem.getConsensusTimestamp(), 3L, tokenId))
+                .receiverAccountId(oldTreasury)
+                .build();
+        var nftTransfer4 = com.hedera.mirror.common.domain.token.NftTransfer.builder()
+                .id(new NftTransferId(nftUpdateRecordItem.getConsensusTimestamp(), 3L, tokenId))
+                .senderAccountId(oldTreasury)
+                .receiverAccountId(newTreasury)
+                .build();
+        assertThat(nftTransferRepository.count()).isEqualTo(4L);
+        assertThat(nftTransferRepository.findAll())
+                .usingRecursiveFieldByFieldElementComparator(RecursiveComparisonConfiguration.builder()
+                        .withComparatorForType(EntityId::compareTo, EntityId.class)
```

Comment:
```suggestion
                        .withComparatorForType(Comparator.nullsFirst(EntityId::compareTo), EntityId.class)
```

I don't know why this doesn't cause NPE consistently.  The problem is there is a `NftTransfer` with `null` `receiverAccountId`, with the EntityId comparator set the original way, it should always trigger an NPE.